assets/scripts/main.js

```diff
@@ -110,4 +110,14 @@
       $('.wcppec-checkout-buttons').hide();
     }
   });
+  
+  // Fixes missing anchor on elementor widget pagination.
+  // Must be combined with <a name="widget"></a> in title.
+  // https://github.com/elementor/elementor/issues/4703
+  $(document).ready(() => {
+    $('.elementor-widget-woocommerce-products').attr('id', $('.elementor-widget-woocommerce-products').attr('data-id'));
```

Comment:
```suggestion
    const $widget = $('.elementor-widget-woocommerce-products');
    if (!$widget.attr('id')) {
      $widget.attr('id', $widget.attr('data-id'));
    }
```
You can use the same `$widget` as context for the selector looking for all links below.